core/array/shared/unshift.rb

```diff
@@ -0,0 +1,47 @@
+describe :array_unshift, shared: true do
+  it "prepends object to the original array" do
+    a = [1, 2, 3]
+    a.send(@method, "a").should equal(a)
+    a.should == ['a', 1, 2, 3]
+    a.send(@method).should equal(a)
+    a.should == ['a', 1, 2, 3]
+    a.send(@method, 5, 4, 3)
+    a.should == [5, 4, 3, 'a', 1, 2, 3]
+
+    # shift all but one element
+    a = [1, 2]
+    a.shift
+    a.send(@method, 3, 4)
+    a.should == [3, 4, 2]
+
+    # now shift all elements
+    a.shift
+    a.shift
+    a.shift
+    a.send(@method, 3, 4)
+    a.should == [3, 4]
+  end
+
+  it "quietly ignores unshifting nothing" do
+    [].send(@method).should == []
+    [].send(@method, *[]).should == []
```

Comment:
RuboCop complains about this line.
I think we should remove it, as it is equivalent to the line above and semantics of splat with an empty array are specified under `language/`.

Reply:
Could you remove that line?